packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/CodeMirror.tsx

```diff
@@ -218,6 +218,13 @@ function CodeMirror(props: NoteBodyEditorProps, ref: any) {
 						textCheckbox: () => addListItem('- [ ] ', _('List item')),
 						textHeading: () => addListItem('## ', ''),
 						textHorizontalRule: () => addListItem('* * *'),
+						'editor.execCommand': (value: any) => {
```

Comment:
Could you create an interface for the value type please? It doesn't have to be exported but it helps as a kind of code documentation. I guess `name` is a required prop, but how about `args`? I guess it shouldn't be required, but if it's not set, is it not going to fail on the function call?

Reply:
That's a good point, my assumption was that if the plugin developer failed to provide `args`, the error message would be pretty self explanatory (and we wouldn't need to clutter this code). However, since this interface is going to be used by both codemirror and tinymce it's possible that this bug will be found by the end user. I'll add some error checking to make it nicer.

This also makes me realize that the one command should be defined and if possible should work with both TinyMCE and CodeMirror. I think I'll change the interface to use specific args. What do you think?
```javascript
await joplin.commands.execute('editor.execCommand', {
        name: 'someCommand',
        codeMirrorArgs: [],
        ui: false, // Optional
        value: {}, // Optional
        tinyMCEargs: {} // Optional
});
```

I don't expect this situation to come up a lot, but it would be nice to be explicit about supporting it.

Reply:
> I'll add some error checking to make it nicer.

I'm not sure, I think it's fine as you had it. I also think it's not needed to add too much error checking. Like if they don't provide the "name", they'll get the "unsupported command" error which is good enough. For the "args" property, I think it should be optional.

```js
if (value.args) {
    editorRef.current[value.name](...value.args);
} else {
    editorRef.current[value.name]();
}
```

But perhaps `...value.args` already work when args is undefined and there's nothing to change?

Reply:
Hmm, I see what you mean for the command syntax but I feel the tinyMCE or CodeMirror prefixes are a bit overkill. It's a pity that the TinyMCE function signature is so strange that there's no mapping between the two editors.

In that case, please keep it like this:

```
{
        name: string;
        args: any; // Should be an array for CodeMirror or an object for TinyMCE
        ui?: boolean, // For TinyMCE only
        value?: any, // For TinyMCE only
}
```

Reply:
Thanks for this @laurent22 I've got it implemented now. I ended up adding the type to `NoteEditor/utils/types.ts` since it's technically used for both CodeMirror and TinyMCE.